src/robotFactory.js

```diff
@@ -1,10 +1,109 @@
 'use strict';
 
-class BaseRobot {}
+const BASIC_COORDS = {
+  x: 0,
+  y: 0,
+  z: 0,
+};
+
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion
+  ) {
+    this.name = name;
+    this.weight = weight;
+
+    this.coords = {
+      x: coords.x || BASIC_COORDS.x,
+      y: coords.y || BASIC_COORDS.y,
+    };
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, `
+      + `Chip version: ${this.chipVersion}, `
+      + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion
+  ) {
+    super(
+      name,
+      weight,
+      coords,
+      chipVersion
+    );
+
+    this.coords.z = coords.z || BASIC_COORDS.z;
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion,
+    maxLoadWeight,
+    currentLoad
+  ) {
+    super(
+      name,
+      weight,
+      coords,
+      chipVersion
+    );
+
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad || null;
+  }
+
+  hookLoad(cargo) {
+    if (this.currentLoad || cargo.weight > this.maxLoadWeight) {
```

Comment:
![image](https://github.com/mate-academy/js_robot_factory/assets/84296041/1ab0afaa-4e24-4fed-8f42-ded5e6d7b04b)

```suggestion
    if (!this.currentLoad && cargo.weight <= this.maxLoadWeight) {
```